Cython/Compiler/Main.py

```diff
@@ -143,6 +143,21 @@ def process_pxd(self, source_desc, scope, module_name):
     def nonfatal_error(self, exc):
         return Errors.report_error(exc)
 
+    def _split_qualified_name(self, qualified_name):
+        # Splits qualified_name into parts in form of 2-tuples: (PART_NAME, IS_PACKAGE).
+        qualified_name_parts = qualified_name.split('.')
+        last_part = qualified_name_parts[-1]
+        qualified_name_parts = [(p, True) for p in qualified_name_parts[:-1]]
+        if last_part != '__init__':
+            # If Last part is __init__, then it is omitted. Otherwise, we need to check whether we can find
+            # __init__.pyx/__init__.py file to determine if last part is package or not.
+            path = self.search_include_directories(qualified_name, suffix='.pyx')
+            if not path:
+                path = self.search_include_directories(qualified_name, suffix='.py')
```

Comment:
This duplication seems wasteful. And it's also not what (I think) we do normally. I think we rather prefer the first that we find, regardless of the type. There probably code for that already.

Reply:
I have created new method for that. I was not able to find any existing method doing this.